src/zkscript/merkle_trees/merkle_tree.py

```diff
@@ -0,0 +1,198 @@
+from tx_engine import Script
+
+
+class MerkleTree:
+    """Class representing a Merkle Tree.
+
+    This class takes a root hash and a specified hash function to create the Merkle Tree
+    structure.
+
+    Attributes:
+        root (str): The root hash of the Merkle Tree. Must be a valid hexadecimal value.
+        hash_function (str): The hash function used in the Merkle Tree. This must be a valid opcode hash function or a
+            combination of valid opcode hash functions.
+        depth (int): The depth (number of levels) of the Merkle tree.
+
+    Methods:
+        __init__(root: str, hash_function: str): Initializes the Merkle Tree. An assertion is
+            performed to ensure the hash function is valid and the root is hexadecimal.
+
+    """
+
+    def __init__(self, root: str, hash_function: str, depth: int):
+        assert all(c in "0123456789abcdefABCDEF" for c in root), f"{root} is not a valid hexadecimal string."
+
+        assert set(hash_function.split(" ")).issubset(
+            {"OP_RIPEMD160", "OP_SHA1", "OP_SHA256", "OP_HASH160", "OP_HASH256"}
+        ), f"{hash_function} is not a valid hash function."
+
+        assert depth > 0
+
+        self.root = root
+        self.hash_function = hash_function
+        self.depth = depth
+
+    def locking_merkle_proof_with_bit_flags(
```

Comment:
What about renaming the functions to locking_merkle_proof and locking_merkle_proof_optimised?